diff --git a/node_modules/@ffmpeg/ffmpeg/dist/esm/classes.js b/node_modules/@ffmpeg/ffmpeg/dist/esm/classes.js
index 9b7684b..65b6a71 100644
--- a/node_modules/@ffmpeg/ffmpeg/dist/esm/classes.js
+++ b/node_modules/@ffmpeg/ffmpeg/dist/esm/classes.js
@@ -102,7 +102,11 @@
     #getWorker = () => {
         if (!this.#worker) {
             this.#worker = classWorkerURL ?
-                new Worker(new URL(classWorkerURL, import.meta.url), {
+                new Worker(new URL(classWorkerURL, typeof document === 'undefined'
+
+					? new URL('file:' + __filename)
+
+					: (document.currentScript && document.currentScript.src) || new URL('main.js', window.location) ), {
                     type: "module",
                 }) :
                 // We need to duplicated the code here to enable webpack
